### PR TITLE
Дженерики платформенных «библиотек» + цепочки Метаданные

### PR DESCRIPTION
**Pull request by sfaqer**
## Summary

- **Платформенные «библиотеки»** — `БиблиотекаКартинок.<Имя картинки>` / `БиблиотекаСтилей.<Имя стиля>` / `БиблиотекаМакетовОформленияКомпоновкиДанных.<Имя макета>` материализуются из `Configuration.getCommonPictures/getStyles/getCommonTemplates` (последняя — с фильтром по `TemplateType.DATA_COMPOSITION_APPEARANCE_TEMPLATE`). Значения enum'а-«библиотеки» получают правильный `returnType` через `ContextEnum.valueType()` из bsl-context (`БиблиотекаКартинок.ИнтервалДат` → `Картинка`, а не сам `БиблиотекаКартинок`).
- **`Метаданные.<коллекция>.<MD>.Реквизиты.<имя>`** — цепочка собирается через новый `MetadataCollectionSpecializer`: Phase B+C из плана (per-MDO стандартные реквизиты из `KnownStandardAttributes`, per-section табчасти с рекурсивной специализацией, override returnType per-MDO по реальному `MDOType` + fallback `instanceof TabularSection`). Источник per-property element-type — bsl-context `ContextProperty.collectionElementTypes()` (Phase A).
- **`TypeRegistry`**: `expandedMembers` (snapshot материализованных members, нужен для self-target expansion), `registerMemberOverride` (addFirst), `memberSources` теперь `CopyOnWriteArrayList` + snapshot в `getMembers` — против CME при параллельной регистрации.
- **`CompletionProvider`**: фильтр generic-template имён через `ContextNames.placeholders` (не пуляем `<Имя X>` в completion-варианты).
- **bump bsl-context** → `0.6.0` (стабильный тег с новым API).

## Test plan

- [x] `MetadataCollectionSpecializerTest` (14) — позитив + edge-cases (composite types, элемент-возвращающие методы, рекурсия по TS).
- [x] `RegisterCommonLibraryExpansionTest` (8) — материализация snapshot'ом, no-op-условия, `appearanceTemplatesOf` фильтр.
- [x] `TypeRegistryExpandedMembersTest` (6) — snapshot без регистрации источника, self-loop case.
- [x] `BslContextPlatformTypesProviderTest` +3 теста (enum-value generic / valueType-returnType / fallback на enumRef).
- [x] `MetadataChainInferenceTest` — интеграционный, на новом API `TypeService.expressionTypesAt` (по реархитектуре #3993), цепочка `Метаданные.Документы.Документ1.Реквизиты.Реквизит1` + ТЧ. Включается `BSL_LANGUAGE_SERVER_RUN_HBK_TESTS=true`.
- [x] `GenericsCoverageDumpTest` — диагностический отчёт по placeholder'ам (живая карта).
- [x] `./gradlew check` локально зелёный.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Generic-template member specialization that materializes concrete members for metadata collections
  * Improved enum-value typing and specialized display names for expanded types

* **Bug Fixes**
  * Completion excludes generic placeholder templates from suggestions

* **Tests**
  * Many new tests for collection specialization, metadata-chain type inference, enum/member typing, and generic-member expansion behavior

### CODE REVIEW

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/MetadataCollectionSpecializer.java [hunk 1]
```diff
@@ -0,0 +1,990 @@
...
+        TypeSet.of(perCollRef),
+        List.of(),
+        null,
+        false,
+        PlatformMetadata.EMPTY,
+        false
+      ));
+    }
+    return overrides;
+  }
```
[the remark this answers is not in the document]
Comment (sfaqer) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/MetadataCollectionSpecializer.java:627]: Skip намеренно не делаю — это сознательное архитектурное решение, прямо описанное в javadoc метода. Override регистрируется **даже когда у владельца нет такой коллекции**, чтобы тип `Метаданные.Документы.X.Команды` стабильно был `КоллекцияОбъектовМетаданных.Команды.X` (а не общий `КоллекцияОбъектовМетаданных`).

> Так пользователь видит, что `Метаданные.Документы.Покупатели.Команды` — это `КоллекцияОбъектовМетаданных.Команды.Покупатели`, а не общий `КоллекцияОбъектовМетаданных`, даже если у документа нет команд.

Если skip — мы теряем consistent type для пустых коллекций и пользователь видит generic-base.

File: src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/GenericsCoverageDumpTest.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    for (var e : genericEnumValuesByOwner.entrySet()) {
+      report.append("- **").append(e.getKey()).append("**\n");
+      for (var v : e.getValue()) {
+        report.append("  - `").append(v).append("`\n");
+      }
+    }
+
+    var outPath = Path.of("D:/tmp/generics-report.md");
+    Files.writeString(outPath, report.toString(), StandardCharsets.UTF_8);
+    System.out.println("Generics report written to " + outPath);
```
[the remark this answers is not in the document]
Comment (sfaqer) [reply src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/GenericsCoverageDumpTest.java]: Файл удалён целиком — это был временный инструмент для подбора ожиданий в `MetadataChainInferenceTest`, в коммит попал по недосмотру.

[the remark this answers is not in the document]
Comment (sfaqer) [reply src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/GenericsCoverageDumpTest.java]: Файл удалён целиком — это был временный инструмент для подбора ожиданий в MetadataChainInferenceTest, в коммит попал по недосмотру.

**Comment by nixel2007** [comment]
@sfaqer мерж конфликт. только аккуратно!

**Comment by nixel2007** [comment]
@sfaqer мерж-конфликт. Внимательно проанализируй изменения в develop и сделай мерж/ребэйз

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/providers/CompletionProvider.java [hunk 3]
```diff
@@ -190,11 +191,11 @@ private boolean libraryEntryVisible(OScriptLibraryIndex.LibraryEntry entry,
       return false;
     }
     var origin = entry.libOrigin();
-    if (origin == null || origin.isBlank()) {
+    if (origin.isBlank()) {
       return true;
     }
     var originLower = origin.toLowerCase(Locale.ROOT);
-    var samePackage = ownLibOrigin.map(originLower::equals).orElse(false);
+    boolean samePackage = ownLibOrigin.isPresent() && originLower.equals(ownLibOrigin.get());
```
Comment (nixel2007) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/providers/CompletionProvider.java]: зачем? это не идеоматичное использование Optional

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationGenericExpander.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+      b.put("Имя таблицы внешнего источника данных", dimTableName);
+      registerFamilySpecializations(familyCore, b);
+    }
+    for (var dim : cube.getDimensions()) {
+      var dimName = dim.getName();
+      if (dimName.isBlank()) {
+        continue;
+      }
+      var b = new LinkedHashMap<>(cubeBindings);
+      b.put("Имя измерения", dimName);
```
Comment (nixel2007) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationGenericExpander.java:122]: оно не развалится при английском ScriptVariant/локали?

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/TypeRegistry.java [hunk 5]
```diff
@@ -369,12 +373,11 @@ private List<MemberDescriptor> computeMembers(TypeRef ref, FileType fileType) {
...
-    // synchronized на самом списке, иначе при конкурентной регистрации возможен CME.
-    List<ScopedMemberSource> snapshot;
-    synchronized (sources) {
-      snapshot = List.copyOf(sources);
-    }
+    // Snapshot: список source'ов может модифицироваться параллельно через
+    // registerMemberSource/registerMemberOverride (Phase B/C MetadataCollectionSpecializer
+    // и др. workspace-scoped провайдеры). sources — CopyOnWriteArrayList,
+    // снимок через List.copyOf дёшев и стабилен на время итерации.
+    List<ScopedMemberSource> snapshot = List.copyOf(sources);
```
Comment (nixel2007) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/TypeRegistry.java:378]: здесь synchronized-блок был для защиты от ConcurrentModificationException. надо откатить.

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProvider.java [hunk 6]
```diff
@@ -436,49 +529,204 @@ private void registerInheritedMembers(TypeRef target, String exactName) {
...
+      .map(Attribute::getName)
+      .filter(n -> !n.isBlank())
+      .toList();
+    if (!names.isEmpty()) {
+      sink.put(placeholder, names);
+    }
+  }
+
+  /** Backward-compat alias для не-static вызовов. */
+  private String memberPlaceholderName(TypeRef generic) {
```
Comment (nixel2007) [inline src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProvider.java]: удалить

File: src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProviderHelpersTest.java [hunk 7]
```diff
@@ -370,18 +370,57 @@ void tryRegister_withInformationRegisterAndGenericTemplate_expandsDimensions() {
 
   @Test
   void tryRegister_withCalculationRegister_invokesRecalculationSpec() {
-    var reg = CalculationRegister.builder().name("Начисления").build();
+    var recalc = com.github._1c_syntax.bsl.mdo.children.Recalculation.builder()
```
Comment (nixel2007) [inline src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProviderHelpersTest.java]: FQN

File: src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProviderHelpersTest.java [hunk 8]
```diff
@@ -370,18 +370,57 @@ void tryRegister_withInformationRegisterAndGenericTemplate_expandsDimensions() {
...
       java.util.Map.of(reg.getMdoReference(), (MD) reg),
       (registry, p) -> {
         p.tryRegister();
         assertThat(registry.resolve("РегистрРасчетаМенеджер.Начисления")).isPresent();
       });
   }
 
+  @Test
+  void tryRegister_withChartOfCalculationTypes_invokesDerived() {
+    var pvr = com.github._1c_syntax.bsl.mdo.ChartOfCalculationTypes.builder()
```
Comment (nixel2007) [inline src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProviderHelpersTest.java]: FQN

**Comment by nixel2007** [comment]
@claude review

File: src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/TypeRegistry.java (hunk 5 above)
Comment (nixel2007) [reply src/main/java/com/github/_1c_syntax/bsl/languageserver/types/registry/TypeRegistry.java:378]: Снимается, тут CopyOnWriteArrayList

File: src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProviderHelpersTest.java
Comment (nixel2007) [inline src/test/java/com/github/_1c_syntax/bsl/languageserver/types/registry/ConfigurationTypesProviderHelpersTest.java:615]: посмотри ВСЕ  FQN в файле

**Comment by nixel2007** [comment]
Я тебе опять конфликт принёс

**Review by nixel2007 — COMMENTED** [review]
@sfaqer тебе нужно сделать ребейз/мерж девелопа, у тебя мерж-конфликт

**Comment by nixel2007** [comment]
@sfaqer javadoc падает
